### PR TITLE
Kh add process rules

### PR DESCRIPTION
**Pull request by Kaizhe**
To test this PR:

1. Checkout the branch

2. Launch falco container with the updated falco rule yaml file

3. Run the following script:
```
#!/bin/bash

set -eux 

client=$(docker run -d --name client kaizheh/nginx-docker)

server=$(docker run -d --name server kaizheh/nginx-docker)

docker exec server bash -c "apt-get update && apt-get -y install nmap"

docker exec server bash -c "nc -lp 1234 -c 'cat /etc/passwd'" &

server_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $server)

docker exec client bash -c "nc $server_ip 1234"
```

To verify the output from falco container:

> 18:22:08.037417144: Error Package management process launched (user=root command=apt-get update container_id=7a222f00f4d0 container_name=server image=kaizheh/nginx-docker)
18:22:09.249714486: Error Package management process launched (user=root command=apt-get -y install nmap container_id=7a222f00f4d0 container_name=server image=kaizheh/nginx-docker)
18:22:19.057577326: Warning Netcat runs inside container that allows remote code execution (user=root command=nc -lp 1234 -c cat /etc/passwd container_id=7a222f00f4d0 container_name=server image=kaizheh/nginx-docker)
18:22:19.174572057: Notice Network tool launched (user=root command=nc 172.17.0.4 1234 container_id=1f2800fba7bc container_name=client image=kaizheh/nginx-docker)
>

### CODE REVIEW

**Comment by Kaizhe** [comment]
Rationale behind adding these rules:
1. package management tool can be used to download utils for malicious purpose (e.g. nmap). Given the micro service container scenario, package management should be done in the image build time for best practice. 
2. Nmap can be used for reconnaissance purpose in customer's enviroment.
3. nc(netcat) and ncat can be either used for port scanning purpose or reverse shell.

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -1720,6 +1726,46 @@
...
 
+- list: network_tool_binaries
+  items: [nc, ncat, nmap]
+
+- macro: network_tool_procs
+  condition: proc.name in (network_tool_binaries)
+
+# Container is supposed to be immutable. Package management should be done in building the image.
+- rule: Launch Package Management Process
+  desc: Package management process ran insdie container
```
Comment (mstemm) [inline rules/falco_rules.yaml]: Typo, insdie -> inside

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -1720,6 +1726,46 @@
...
   tags: [network, k8s, container]
 
+- list: network_tool_binaries
+  items: [nc, ncat, nmap]
+
+- macro: network_tool_procs
+  condition: proc.name in (network_tool_binaries)
+
+# Container is supposed to be immutable. Package management should be done in building the image.
+- rule: Launch Package Management Process
```
Comment (mstemm) [inline rules/falco_rules.yaml]: Maybe add "in Container" to the rule (and later rules), so it's clear that package manage procs on the host are okay?

File: rules/falco_rules.yaml [hunk 3]
```diff
@@ -1720,6 +1726,46 @@
...
+  desc: Package management process ran insdie container
+  condition: >
+    spawned_process and container and user.name != "_apt" and package_mgmt_procs and not package_mgmt_ancestor_procs
+  output: >
+    Package management process launched (user=%user.name
```
Comment (mstemm) [inline rules/falco_rules.yaml]: Again, an "in container" would prevent any confusion.

**Pull request merged**